File: 2551-apply-operations-to-an-array/apply-operations-to-an-array.py
class Solution:
    def applyOperations(self, nums: List[int]) -> List[int]:
        n=len(nums)
        for i in range(n-1):
            if nums[i]==nums[i+1]:
                nums[i]*=2
                nums[i+1]=0

        pos=0
        ls=[0]*n
        for item in nums:
            if item!=0:
                ls[pos]=item
                pos+=1
        return ls       

File: 2551-apply-operations-to-an-array/apply-operations-to-an-array.ts
function applyOperations(nums: number[]): number[] {
    
    let j=0;
    let n = nums.length
    for (let i=0;i<n-1;i++){
        if (nums[i]!=0){
            if (nums[i]==nums[i+1]){
                nums[j]=nums[i]*2
                nums[i+1]=0
            }
            else{
                nums[j]=nums[i]
            }
            j+=1
        }
        
    }
    if (nums[n-1]!=0){
        nums[j]=nums[n-1]
        j+=1
    }
    while(j<n){
        nums[j]=0
        j+=1
    }
    
    return nums
};